src/main/java/pl/codewise/internship/MessageBuffer.java

```diff
@@ -0,0 +1,41 @@
+package pl.codewise.internship;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+import java.util.concurrent.locks.ReadWriteLock;
+import java.util.concurrent.locks.ReentrantReadWriteLock;
+import java.util.stream.Collectors;
+
+public final class MessageBuffer{
```

Comment:
Wyciągnięte. Wyciągnąłem też od razu `MessageWrapper`, bo skoro dwie klasy z niego korzystają to też chyba powinien być na zewnątrz.